src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/11.diff.sql

```diff
@@ -0,0 +1,150 @@
+
+END
+GO
+
+/*************************************************************
+    Stored procedure for adding a workitem.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     AddWorkitemV11
+--
+-- DESCRIPTION
+--     Adds a UPS-RS workitem.
+--
+-- PARAMETERS
+--     @partitionKey
+--         * The system identifier of the data partition.
+--     @workitemUid
+--         * The workitem UID.
+--     @stringExtendedQueryTags
+--         * String extended query tag data
+--     @dateTimeExtendedQueryTags
+--         * DateTime extended query tag data
+--     @personNameExtendedQueryTags
+--         * PersonName extended query tag data
+--     @initialStatus
+--         * New status of the workitem, Either 0(Creating) or 1(Created)
+-- RETURN VALUE
+--     The WorkitemKey
+--     The Watermark
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.AddWorkitemV11
+    @partitionKey                   INT,
+    @workitemUid                    VARCHAR(64),
+    @stringExtendedQueryTags        dbo.InsertStringExtendedQueryTagTableType_1 READONLY,
+    @dateTimeExtendedQueryTags      dbo.InsertDateTimeExtendedQueryTagTableType_2 READONLY,
+    @personNameExtendedQueryTags    dbo.InsertPersonNameExtendedQueryTagTableType_1 READONLY,
+    @initialStatus                  TINYINT
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+    DECLARE @watermark BIGINT
+    DECLARE @workitemKey BIGINT
+    DECLARE @currentDate DATETIME2(7) = SYSUTCDATETIME()
+
+    SELECT @workitemKey = WorkitemKey
+    FROM dbo.Workitem
+    WHERE PartitionKey = @partitionKey
+        AND WorkitemUid = @workitemUid
+
+    IF @@ROWCOUNT <> 0
+        THROW 50409, 'Workitem already exists', 1;
+
+    -- The workitem does not exist, insert it.
+    SET @workitemKey = NEXT VALUE FOR dbo.WorkitemKeySequence
+    SET @watermark = NEXT VALUE FOR dbo.WorkitemWatermarkSequence
+    INSERT INTO dbo.Workitem
+        (WorkitemKey, PartitionKey, WorkitemUid, Status, Watermark, CreatedDate, LastStatusUpdatedDate)
+    VALUES
+        (@workitemKey, @partitionKey, @workitemUid, @initialStatus, @watermark, @currentDate, @currentDate)
+
+    BEGIN TRY
+
+        EXEC dbo.IIndexWorkitemInstanceCore
+            @partitionKey,
+            @workitemKey,
+            @stringExtendedQueryTags,
+            @dateTimeExtendedQueryTags,
+            @personNameExtendedQueryTags
+
+    END TRY
+    BEGIN CATCH
+
+        THROW
+
+    END CATCH
+
+    COMMIT TRANSACTION
+
+    SELECT
+        @workitemKey,
+        @watermark
+
+END
+GO
+
+COMMIT TRANSACTION
+
+DROP INDEX IF EXISTS IX_Workitem_WorkitemUid_PartitionKey ON dbo.Workitem
+GO
+
+CREATE UNIQUE NONCLUSTERED INDEX IX_Workitem_WorkitemUid_PartitionKey ON dbo.Workitem
+(
+    WorkitemUid,
+    PartitionKey
+)
+INCLUDE
+(
+    Watermark,
+    WorkitemKey,
+    Status,
+    TransactionUid
+)
+WITH (DATA_COMPRESSION = PAGE)
```

Comment:
Add DROP_EXISTING = ON instead of deleting separately

Reply:
Even this is fine

Reply:
I think DROP IF EXISTS should be good? we can change it, if you still insist :)

Reply:
Fine with me